### PR TITLE
GH-63: Fix CodeCov, add Unit Tests

### LINKED ISSUE #63: Create Unit Tests
We want to create unit tests to increase code coverage.

### PR DESCRIPTION
**Pull request by tiemonl**
Closes #63

### CODE REVIEW

**Comment by tiemonl** [comment]
Some notes on this PR.

- jacoco/testDebugUnitTest.exec was never getting created therefore tests couldn't be read. To solve this, introducing [Android-JUnit5](https://github.com/mannodermaus/android-junit5) plug in alleviated that issue.
- jacocoTestReport provides multiple reports. Returning the report `app` instead of `debugAndroidTest` then showed us our actual code coverage in jacoco.
  - As a result of this, the circleci report path needed to be changed as well.

File: app/src/main/kotlin/io/imhungry/common/ui/BaseActivity.kt [hunk 1]
```diff
@@ -5,13 +5,15 @@ import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
 import com.google.firebase.auth.FirebaseAuth
 import dagger.android.AndroidInjection
+import io.imhungry.login.AuthFailureCallback
 import io.imhungry.login.handleLoginActivityResult
+import io.imhungry.login.launchLoginActivity
 
 abstract class BaseActivity : AppCompatActivity(), FirebaseAuth.AuthStateListener {
 
-    protected var userUnauthenticatedHandler: (() -> Unit)? = null
```
Comment (TripleFrequency) [inline app/src/main/kotlin/io/imhungry/common/ui/BaseActivity.kt]: Why did you remove this? It should also be of type `AuthFailureCallback?`, with a default of null.

File: app/src/main/kotlin/io/imhungry/common/ui/BaseActivity.kt [hunk 2]
```diff
@@ -32,12 +34,12 @@ abstract class BaseActivity : AppCompatActivity(), FirebaseAuth.AuthStateListene
 
     override fun onAuthStateChanged(auth: FirebaseAuth) {
         if (auth.currentUser == null) {
-            userUnauthenticatedHandler?.invoke()
+            launchLoginActivity()
```
Comment (TripleFrequency) [inline app/src/main/kotlin/io/imhungry/common/ui/BaseActivity.kt]: Again, you should use the `userUnauthenticatedHandler` still.
